legacy/builder/test/helper_tools_downloader.go

```diff
@@ -454,30 +454,17 @@ func downloadAndUnpackCore(core Core, url string, targetPath *paths.Path) error
 	packagerPath := targetPath.Join(core.Maintainer)
 	corePath := targetPath.Join(core.Maintainer, core.Arch)
 
+	if err := packagerPath.MkdirAll(); err != nil {
+		return errors.WithStack(err)
+	}
 	if corePath.Exist() {
 		if err := corePath.RemoveAll(); err != nil {
 			return errors.WithStack(err)
 		}
 	}
-
-	if len(files) == 1 && files[0].IsDir() {
-		if err := packagerPath.MkdirAll(); err != nil {
-			return errors.WithStack(err)
-		}
-		if err := unpackFolder.Join(files[0].Name()).CopyDirTo(targetPath.Join(core.Maintainer, core.Arch)); err != nil {
-			return errors.WithStack(err)
-		}
-	} else {
-		if err := targetPath.Join(core.Maintainer, core.Arch).MkdirAll(); err != nil {
-			return errors.WithStack(err)
-		}
-		for _, file := range files {
-			if err := unpackFolder.Join(file.Name()).CopyDirTo(targetPath.Join(core.Maintainer, core.Arch, file.Name())); err != nil {
-				return errors.WithStack(err)
-			}
```

Comment:
Actually, I didn't re-check this part very carefully, because it is used in these legacy integration tests that we will not expand further (and we will eventually migrate to our main integration test framework). There are some code paths that are never visited, this is probably one of them.

So my rationale was: the legacy tests still pass => it's OK :-)